include/eve/module/real/algorithm/function/regular/generic/reduce.hpp

```diff
@@ -9,57 +9,40 @@
 
 #include <eve/concept/vectorized.hpp>
 #include <eve/detail/implementation.hpp>
-#include <eve/detail/is_native.hpp>
-#include <eve/function/combine.hpp>
+#include <eve/detail/function/reduce.hpp>
+#include <eve/function/logical_and.hpp>
+#include <eve/function/logical_or.hpp>
+#include <eve/function/minimum.hpp>
+#include <eve/function/maximum.hpp>
 #include <eve/function/splat.hpp>
-#include <eve/function/swap_adjacent_groups.hpp>
-#include <eve/pattern.hpp>
-#include <bit>
+#include <eve/function/all.hpp>
+#include <eve/function/any.hpp>
 
 namespace eve::detail
 {
-  //================================================================================================
-  // reduce toward wide
   template<real_scalar_value T, typename N, typename ABI, typename Callable>
   EVE_FORCEINLINE auto reduce_( EVE_SUPPORTS(cpu_), splat_type const&
                               , wide<T,N,ABI> v, Callable f
                               ) noexcept
   {
-    if constexpr( N::value == 1 )
-    {
-      return v;
-    }
-    else if constexpr( !is_aggregated_v<ABI> )
-    {
-      //============================================================================================
-      // We have exactly log2(cardinal) - 1 butterfly steps to perform
-      constexpr auto depth = std::bit_width(std::size_t{N::value}) - 1;
+          if constexpr( std::same_as<Callable, callable_min_> ) return splat(minimum)(v);
+    else  if constexpr( std::same_as<Callable, callable_max_> ) return splat(maximum)(v);
```

Comment:
Why not just compute the functor? Then you don't need to do it twice

Reply:
will do